### ISSUE TITLE
Don't bind initially hidden (not visible) StateNode and bind it once it becomes visible

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/flow#3148]
Part of the #3143

Related to the #3147.
But this is another topic.

We send default property values from the client-side to the server side if they are not overridden by the server. It becomes an issue when the component is "hidden" from the very beginning. In that case since we are preventing the RPCs then those default values won't be sent to the server. But that's bad. Since the component may become "visible" at some point and its model values will be not in sync with the client. To be able to solve this issue I suggest _DO NOT BIND_ the initially hidden StateNode. The same thing will be used : use a fake DOM node to represent it in the DOM tree but don't create any real element and don't save it in the StateNode instance. Postpone this when the element will be "visible" and bind it at this point. It might be that this will require information about whether the node is bound or not. This can be done via additional key in the visibility NodeMap (in the same way as with the fake node).

It will require some implementation API for `StateNode` like "isBound" .
Not sure that it should go into the `StateNode` class.
We may mark it as not bound as described above using the feature and a property in the feature.
Another way (to check): if the `StateNode` has no `domNode` set but it's attached.
This may collide with the attach existing elements functionality ( by id or subtemplates).
Those nodes may have `null` for `domNode`  initially (until the web component element become initialized/ready).

This depends on #3146.

**Comment by pleku** [comment]
Just a reminder that there are several cases to test here with templates, subtemplates and using id-mapping.

**Comment by denis-anisimov** [comment]
Unfortunately it's not possible to avoid element creation.
Because the parent expects a DOM element available for a child `StateNode`.

So I just have to create some fake node in this case instead of the real one (and that why actually I suggested to create a fake node even in all cases instead of hiding it: to use a unified solution).
But this fake node won't be regularly bound.
Once the node becomes visible the fake node will be unbound, replaced with the real node which will be created from scratch and rebound to the `StateNode`.
